src/gt4sd/frameworks/crystals_rfc/FeatureEngine.py

```diff
@@ -0,0 +1,223 @@
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+"""Feature engine for crystals."""
+
+import os
+import re
+from typing import Any, Dict, List
+
+import pandas as pd
+import pkg_resources
+from pymatgen.core.composition import Composition
+
+
+class Features:
+    """Feature generator for crystals."""
+
+    def __init__(self, formula_file: str):
+        """Initialize Feature engine.
+
+        Args:
+            formula_file: file of formulas
+        """
+        self.formula_file = formula_file
+        self.atomic_data_file = pkg_resources.resource_filename(
+            "gt4sd", os.path.join("frameworks", "crystals_rfc", "atomic_data.csv")
+        )
+
+    def make_features(
+        self,
+        atomic_descriptors,
+        formula_list,
+        targets,
+        encoded_sym,
+        addAVG: bool = True,
+        addAAD: bool = True,
+        addMD: bool = False,
+        addCV: bool = False,
+    ) -> List[List[Any]]:

```

Comment:
not sure I get all types right, but this Any could maybe be a Union of some types?